src/node_report_utils.cc

```diff
@@ -7,15 +7,42 @@ using node::MallocedBuffer;
 
 static constexpr auto null = JSONWriter::Null{};
 
+// Utility function to format socket information.
+static bool ReportEndpoint(uv_handle_t* h,
+                           struct sockaddr* addr,
+                           const char* name,
+                           JSONWriter* writer) {
+  uv_getnameinfo_t endpoint;
+  if (uv_getnameinfo(h->loop, &endpoint, nullptr, addr, NI_NUMERICSERV) == 0) {
+    writer->json_objectstart(name);
+    writer->json_keyvalue("host", endpoint.host);
+    writer->json_keyvalue("port", endpoint.service);
+    writer->json_objectend();
+    return true;
+  } else {
+    char host[INET6_ADDRSTRLEN];
+    int family = addr->sa_family;
+    if (uv_inet_ntop(family, addr, host, sizeof(host)) == 0) {
+      std::string port = std::to_string(ntohs(family == AF_INET ?
+                         reinterpret_cast<sockaddr_in*>(addr)->sin_port :
+                         reinterpret_cast<sockaddr_in6*>(addr)->sin6_port));
```

Comment:
Does it make sense to represent the port as an integer? That would allow skipping the stringifying step, but I imagine we’d also need to do that in the `uv_getnameinfo()` case?

Reply:
Yes, it probably does make sense to be an integer. I've stringified it here for consistency with the existing `uv_getnameinfo()` case.

Reply:
If you just assign the host and port in the conditionals, then you can DRY the writer code a bit.